### DISCUSSION TITLE
Pipe archive

### DISCUSSION THREAD

**Discussion by florianbepunkt** [101arrowz/fflate#206] [Q&A]
Given the following example

```ts
const zip = new Zip();

const helloTxt = new ZipDeflate("hello.txt", { level: 9 });
zip.add(helloTxt);
helloTxt.push(Buffer.from("hello"));
helloTxt.push(Buffer.from(" "));
helloTxt.push(Buffer.from("world"));
zip.end();
```

How can I pipe the `zip` stream into the next step of my pipeline? It does not conform to a duplex stream interface.

**Comment by 101arrowz** [comment]
You can write to the next stream in the chain manually. For Node.js:

```js
zip.ondata = (err, chunk, final) => {
  if (err) writable.destroy(err);
  else {
    writable.write(chunk);
    if (final) writable.end();
  }
}
```

For the WHATWG web streams API, have a look at [this guide](https://github.com/101arrowz/fflate/wiki/Guide:-Modern-(Buildless)); midway through is an example to convert an `fflate` stream to a `ReadableStream`.
